x/staging/posts/keeper/grpc_query.go

```diff
@@ -16,98 +15,38 @@ import (
 
 var _ types.QueryServer = Keeper{}
 
-func (k Keeper) getPostResponse(ctx sdk.Context, post types.Post) types.QueryPostResponse {
-	// Get the reactions
-	postReactions := k.GetPostReactions(ctx, post.PostID)
-	if postReactions == nil {
-		postReactions = []types.PostReaction{}
-	}
-
-	// Get the children
-	childrenIDs := k.GetPostChildrenIDs(ctx, post.PostID)
-	if childrenIDs == nil {
-		childrenIDs = []string{}
-	}
-
-	//Get the user answers if poll exist
-	var answers []types.UserAnswer
-	if post.PollData != nil {
-		answers = k.GetUserAnswersByPost(ctx, post.PostID)
-	}
-
-	// Crete the response object
-	return types.QueryPostResponse{
-		Post:        post,
-		UserAnswers: answers,
-		Reactions:   postReactions,
-		Children:    childrenIDs,
-	}
-}
-
+// Posts implements the Query/Posts gRPC method
 func (k Keeper) Posts(goCtx context.Context, req *types.QueryPostsRequest) (*types.QueryPostsResponse, error) {
-	var filteredPosts []types.QueryPostResponse
+	var posts []types.Post
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
 	store := ctx.KVStore(k.storeKey)
-	postsStore := prefix.NewStore(store, types.PostStorePrefix)
+	postsStore := prefix.NewStore(store, types.PostQuerySubspacePrefix(req.Subspace))
 
 	pageRes, err := query.FilteredPaginate(postsStore, req.Pagination, func(key []byte, value []byte, accumulate bool) (bool, error) {
+		store := ctx.KVStore(k.storeKey)
+		bz := store.Get(types.PostStoreKey(string(value)))
+
 		var post types.Post
-		if err := k.cdc.UnmarshalBinaryBare(value, &post); err != nil {
+		if err := k.cdc.UnmarshalBinaryBare(bz, &post); err != nil {
 			return false, status.Error(codes.Internal, err.Error())
 		}
 
-		matchParentID, matchCreationTime, matchSubspace, matchCreator, matchHashtags := true, true, true, true, true
-
-		// match parent id if valid
-		if types.IsValidPostID(req.ParentId) {
-			matchParentID = req.ParentId == post.ParentID
-		}
-
-		// match creation time if valid height
-		if req.CreationTime != nil {
-			matchCreationTime = req.CreationTime.Equal(post.Created)
-		}
-
-		// match subspace if provided
-		if req.Subspace != "" {
-			matchSubspace = req.Subspace == post.Subspace
-		}
-
-		// match creator address (if supplied)
-		if req.Creator != "" {
-			matchCreator = req.Creator == post.Creator
-		}
-
-		// match hashtags if provided
-		if req.Hashtags != nil {
-			postHashtags := post.GetPostHashtags()
-			matchHashtags = len(postHashtags) == len(req.Hashtags)
-			sort.Strings(postHashtags)
-			sort.Strings(req.Hashtags)
-			for index := 0; index < len(req.Hashtags) && matchHashtags; index++ {
-				matchHashtags = postHashtags[index] == req.Hashtags[index]
-			}
-		}
-
-		if matchParentID && matchCreationTime && matchSubspace && matchCreator && matchHashtags {
-			if accumulate {
-				filteredPosts = append(filteredPosts, k.getPostResponse(ctx, post))
-			}
-
-			return true, nil
+		if accumulate {
+			posts = append(posts, post)
 		}
 
-		return false, nil
+		return true, nil
```

Comment:
Why was this changed? Shouldn't it be kept to `false` since it represents the `stop`?

Reply:
@RiccardoM  This is for [`FilteredPaginate`](https://github.com/cosmos/cosmos-sdk/blob/9fd866e3820b3510010ae172b682d71594cd8c14/types/query/filtered_pagination.go#L91), `true` here represents that the result is hit to count the number of element for paginated.

Reply:
@dadamu Oh ok thanks. Great